src/main/java/edu/kis/powp/jobs2d/command/ImmutableComplexCommand.java

```diff
@@ -8,10 +8,13 @@
 import edu.kis.powp.jobs2d.Job2dDriver;
 
 public class ImmutableComplexCommand implements ICompoundCommand {
+
     private List<DriverCommand> driverCommandList;
+    private String commandName;
 
-    public ImmutableComplexCommand(List<DriverCommand> driverCommandList) {
+    public ImmutableComplexCommand(List<DriverCommand> driverCommandList, String commandName) {
```

Comment:
It would be safer to add (not modify) the constructor. The old one might be in use by other (work in progress) features. You can add a default commandName (e.g. "unknown immutable command")

Reply:
Done